austation/code/modules/hydroponics/plant_genes.dm

```diff
@@ -34,3 +34,38 @@
 /datum/plant_gene/trait/noreact/on_squashreact(obj/item/reagent_containers/food/snacks/grown/G, atom/target)
 	DISABLE_BITFIELD(G.reagents.flags, NO_REACT)
 	G.reagents.handle_reactions()
+
+//Too close to Neck-Code TM
+/datum/plant_gene/trait/spines
+	name = "Floral Spines"
+
+/datum/plant_gene/trait/spines/on_slip(obj/item/reagent_containers/food/snacks/grown/G, atom/target)
+	if(G.seed)
+		if(G.seed.get_gene(/datum/plant_gene/trait/spines))
+			if(istype(target, /mob/living))
+				G.target = target
+
+				G.grown_overlay.layer = FLOAT_LAYER
+				G.target.add_overlay(G.grown_overlay, TRUE)
+			
+				var/P = G.seed.get_gene(/datum/plant_gene/trait/stinging)
+				var/mob/living/L = target
+				L.adjustBruteLoss((G.seed.potency/4.7)*P)
+
+				G.tryEmbed(target, TRUE, TRUE)
+
+/datum/plant_gene/trait/spines/on_throw_impact(obj/item/reagent_containers/food/snacks/grown/G, atom/target)
+	if(!..())
+		if(G.seed)
+			if(G.seed.get_gene(/datum/plant_gene/trait/spines))
+				if(istype(target, /mob/living))
+					G.target = target
+
+					G.grown_overlay.layer = FLOAT_LAYER
+					G.target.add_overlay(G.grown_overlay, TRUE)
+				
+					var/P = G.seed.get_gene(/datum/plant_gene/trait/stinging)
+					var/mob/living/L = target
+					L.adjustBruteLoss((G.seed.potency/4.7)*P)
```

Comment:
this is going to runtime if the plant doesn't have the stinging gene, as P will be null